pages/studentPortal/labs/multiplication-connect/Index.tsx

```diff
@@ -87,16 +69,60 @@ export const calculateWinner = (grid: GameBoardBlock[]) => {
 };
 
 const Index: FC = () => {
-  //Extractd to App level State to determine Winner here
-  const [grid, setGrid] = useState(createGrid);
+  const [grid, setGrid] = useState([]);
+  const [newGame, setNewGame] = useState(0);
+
+  // initialize grid on page load
+  const createGrid = () => {
+    let arr = [];
+    let newGrid = [];
+    for (let i = 4; i < 25; i++) i % 2 === 0 ? arr.push(i) : "";
+    for (let i = 0; i < 35; i++) {
+      let gridNumber = getRandomItemFromArray(arr);
+      newGrid.push({
+        id: i,
+        gridNumber: gridNumber,
+        isSelected: false,
+      });
+    }
+    console.log("Grid created");
+    console.log(newGrid);
+    return newGrid;
+  };
+
+  useEffect(() => {
+    console.log(`newGame: ${newGame}`);
+    setGrid(createGrid);
```

Comment:
I think your effect is set up correctly. But should you be invoking the createGrid function here?

createGrid()

Reply:
I've tried this but it isn't working as intended. Could you help with debugging it when you can?

Reply:
I won't be able to. Try booking Abhi or Jason